utility/doc/PROTOCOL_CLIENT_SIDE_CHALLENGE.md

```diff
@@ -0,0 +1,213 @@
+nodes forms an ***eventually consistent*** system in aggregate. At worst
+however, depending on the underlying logic of the exposed blockchain API
+method, its likely that the session offers ***no consistency***.
+
+### Client Side Challenge Requirements
+
+At minimum, client side challenges can only occur on methods that are
+***idempotent*** and ***eventually consistent*** across a group of nodes.
+
+The underlying response being ***eventually consistent*** means that nodes who
+respond with an error response that indicates the desired resource could not be
+found must be assumed to be operating as expected without further logic
+defining SLA expectations of individual methods.
+
+The **idempotency** requirement of the challenge is to prevent the following
+situations from happening when a node is operating in good faith:
+
+- DoS through chain bloat from an app submitting thousands of valid challenge
+  transactions.
+- Attacks on other node stakes in the situation that client side challenges
+  result in a net loss for the nodes who fail them.
+- Illegitimate reward farming for apps in the situation that client side
+  challenges result in a net gain for the app that executes the challenge.
+
+Additionally, to remain non-interactive, it needs to be the case that the
+relay request is inferable from the result of the relay response.
+
+### Client Side Challenge Mechanism
+
+At the time of whitelisting a relay chain for rewards, it will be required to
+able provide a pattern string that will pattern match to identify if a request
+is unable to be client-side challenged.
+
+This will require that the `pocketcore/SupportedBlockchains` parameter take
+the following form:
+
+```go
+type SupportedBlockchains struct {
+   Blockchains []SupportedBlockchain
+}
+
+type SupportedBlockchain struct {
+    RelayChainID string
+    NotChallengablePatterns []string
+}
+```
+
+The `NotChallengablePatterns` field is an array of pattern strings that should
+successfully match against the raw request string should if the request is not
+able to be client side challenged. This means having to make a change to what
+is expected of the app when requesting a Relay so that the App will send along
+a signed version of the string encoded HTTP request.
+
+```go
+type QueryRelayRequest stuct {
+    Payload       RelayPayload
+    SignedPayload string
+    Meta          RelayMetadata
+    Proof         RelayProof
+}
+```
+
+Then what is expected of the node when responding to a relay is the following
+`RelayResponse`, where the `Signature` field is now the `Signature` of the
+App's original signed payload appended with the `Response` string.
+
+```go
+type RelayResponse struct {
+	Signature string
+    Request   string
+	Response  string
+	Proof     RelayProof
+}
+```
+
+From here, the client side challenge is to be initiated from a transaction
+message sent with the following type:
+
+```go
+type ChallengeProofInvalidData struct {
+	MajorityResponses []RelayResponse
```

Comment:
1. Please add comments to fields (e.g. `ReporterAddress is the staked Application`)
2. It is not clear to me how `MajorityResponses` was aggregated. FOr example:
- Does the application do it?
- Does the app do it every time?
- Should we consider adding a diagram similar to what's in https://forum.pokt.network/t/client-side-validation/148?